CONTRIBUTING.md

```diff
@@ -3,6 +3,37 @@
 We'd love your help! Please join our weekly [SIG
 meeting](https://github.com/open-telemetry/community#special-interest-groups).
 
+## Target audiences
+
+The OpenTelemetry Collector has two main target audiences:
+
+1. End-users, aiming to use an OpenTelemetry Collector binary as a middleware on their infra, sitting between their
+workloads and the telemetry data storage.
+1. Downstream distributions, which can be an official one as part of the OpenTelemetry Collector project, such as the
+"core" and "contrib", or external distributions, such as other open-source projects building on top of the collector or
+vendor-specific distributions.
+
+### End-users
+
+End-users are the target audience for our binary distributions, as made available via the
+[opentelemetry-collector-releases](https://github.com/open-telemetry/opentelemetry-collector-releases) repository. To
+them, stability in the behavior is important, be it runtime or configuration. They are more numerous and harder to get
+in touch with, making our changes to the collector more disruptive to them than to other audiences. As a general rule,
+whenever you are developing OpenTelemetry Collector components (extensions, receivers, processors, exporters), you
+should have end-users' interests in mind. Similarly, changes to code within packages like `config` will have an impact
+on this audience. Make sure to cause minimal disruption when doing changes here.
+
+### Downstream distributions
+
+In this capacity, the opentelemetry-collector repository acts as an API for other projects. This audience cares mostly
```

Comment:
Maybe be specific about what API you are talking about:

`the opentelemetry-collector repository's public Go types, functions, and interfaces act as an API for other projects`